server/rosetta/lib/internal/service/data.go

```diff
@@ -69,8 +72,17 @@ func (on OnlineNetwork) Block(ctx context.Context, request *types.BlockRequest)
 		if err != nil {
 			return nil, errors.ToRosetta(err)
 		}
-	default:
-		err := errors.WrapError(errors.ErrBadArgument, "at least one of hash or index needs to be specified")
+	default: // both empty
+		blockResponse, err = on.client.BlockTransactionsByHeight(ctx, nil)
+		if err != nil {
+			return nil, errors.ToRosetta(err)
+		}
+	}
+
+	// Both of index and hash can be specified in reuqest, so make sure they are not mismatching.
+	if (request.BlockIdentifier.Index != nil && *request.BlockIdentifier.Index != blockResponse.Block.Index) ||
+		(request.BlockIdentifier.Hash != nil && *request.BlockIdentifier.Hash != blockResponse.Block.Hash) {
+		err := errors.WrapError(errors.ErrBadArgument, "mismatching index and hash")
```

Comment:
```suggestion
		err := errors.WrapError(errors.ErrBadArgument, "mismatching index or hash")
```

Reply:
Honestly, we should probably break this up into two different `if` cases, one for index and one for hash and return an error unique to that.

E.g.

```go
if request.BlockIdentifier.Index != nil && *request.BlockIdentifier.Index != blockResponse.Block.Index {
  err := errors.WrapErrorf("mismatched index; expected: %d, got: %d", ...)
}
```